### DISCUSSION TITLE
Guardian/Admin UI password lockout/lost password

### DISCUSSION THREAD

**Discussion by EthnTuttle** [fedimint/fedimint#2826] [General]
While reviewing [this PR](https://github.com/fedimint/ui/pull/113/), I wanted to bring up some discussion around the Guardian/Admin UI password.

1. Is there a recovery method? If not, could there ever be a way where a super majority of Guardians are able to recover said password? (I suppose this could be accomplished by a password manager with recovery as a module, but that seems problematic. Storing a secret in the system that the secret is for.)
2. Account lock out. Perhaps a bad password multiple times could trigger a lockout with exponential increase (I think of my phone that will lock it self for x minutes after y number of failed logins.) I imagine this to be a setting for individual Guardians or Federations. i.e. A federation requires Guardians to have certain robustness of security settings configured.
3. Password rotation?  I believe this is just an authenticated API call? (but I'm not certain so am noting here as a consideration)

It seems like, currently, a naive approach would be spinning up a second federation and moving everything over, but this is not ideal, imho.

**Comment by justinmoon** [comment]
Some notes on how passwords are managed currently in `fedimintd`:

* After configs are generated via config API, the config password is written to disk [here](https://github.com/fedimint/fedimint/blob/127ee6aa20760bf87c6aa2cca5974e81cd46ce15/fedimint-server/src/config/api.rs#L285)
* If `FM_PASSWORD` environment variable is set when `fedimintd` first runs, [this password will be used for the config api](https://github.com/fedimint/fedimint/blob/127ee6aa20760bf87c6aa2cca5974e81cd46ce15/fedimint-server/src/lib.rs#L123-L136). 
* Whenever `FM_PASSWORD` is set, it will overwrite any previous password that was written to disk [here](https://github.com/fedimint/fedimint/blob/master/fedimintd/src/fedimintd.rs#L239-L244). 
* This [plaintext password file](https://github.com/fedimint/fedimint/blob/127ee6aa20760bf87c6aa2cca5974e81cd46ce15/fedimint-server/src/config/io.rs#L36) is how we persist passwords. If we just stored a password hash, we wouldn't be able to handle automatic restarts of fedimintd because configs are encrypted with this password.

One weird error state you can get into now with the setup UI is that if `FM_PASSWORD` environment variable is set when it is initially run, the leader can't set a password in the first screen. Obviously, we could hide password input if it's already been set on the server. But I'm tempted to suggest we just get rid of `FM_PASSWORD` for now because it really complicates things and doesn't do anything useful. Instead, password can only be set via config API.

<img width="324" alt="image" src="https://github.com/fedimint/fedimint/assets/4335621/26167ccc-309a-4250-8981-81dd91cc1bc9">
